### PR TITLE
Ad expense model and specs

### PR DESCRIPTION
**Pull request by apalski**
PR for adding the Expense model, Expenses table and Expense model specs.
Also:
- add Expense factory
- add generic error messages in config/locale/en.yml for English speaking users
- use factory and shoulda matchers in spec
- ran Rubocop and fixed violations

Looking for feedback on specs, specifically on the use of the factory and shoulda matchers. Also, looking for feedback on any areas that need improvement.
Thank you!

### CODE REVIEW

**Comment by apalski** [comment]
Hi @elle 
I think this PR is ready for review. Thanks again for your help!

File: config/locales/en.yml [hunk 1]
```diff
@@ -27,3 +27,9 @@ en:
 
   titles:
     application: Budgetapp
+
+  activerecord:
+    errors:
+      messages:
+        blank: "can't be blank"
+        taken: "already taken"
```
Comment (elle) [inline config/locales/en.yml]: This change is also in another PR. A suggestion to rebase this once the other PR is merged (or vice versa)

File: db/migrate/20180120042303_create_expenses.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateExpenses < ActiveRecord::Migration[5.1]
+  def change
+    create_table :expenses do |t|
+      t.string :name, null: false
+      t.date :due_date
+      t.string :frequency, null: false
+      t.decimal :bill_estimate
```
Comment (elle) [inline db/migrate/20180120042303_create_expenses.rb]: Missing timestamps

File: db/schema.rb [hunk 3]
```diff
@@ -1,7 +1,19 @@
-ActiveRecord::Schema.define(version: 20180115215649) do
+ActiveRecord::Schema.define(version: 20180120042303) do
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
 
+  create_table "budgets", force: :cascade do |t|
+    t.string "name", null: false
+    t.string "budget_type", default: "basic"
+  end
```
Comment (elle) [inline db/schema.rb]: Budget doesn't belong here

File: spec/factories.rb [hunk 4]
```diff
@@ -1,2 +1,6 @@
 FactoryBot.define do
+  factory :expense do
+    sequence(:name) { |n| "name#{n}" }
+    sequence(:frequency) { |n| "frequency#{n}" }
```
Comment (elle) [inline spec/factories.rb]: Since frequency is not unique, a suggestion to use a value that is closer to real life

File: app/models/expense.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class Expense < ActiveRecord::Base
+
+  validates :name, :frequency, presence: true
+  validates :name, uniqueness: true
```
Comment (elle) [inline app/models/expense.rb]: A question: why does name need to be unique?

**Pull request merged**